test/integration/auth/svcaccttoken_test.go

```diff
@@ -79,6 +83,12 @@ func TestServiceAccountTokenCreate(t *testing.T) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	// Enable the node token improvements feature gates prior to starting the apiserver, as the node getter is
+	// conditionally passed to the service account token generator based on feature enablement.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceAccountTokenNodeBinding, true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceAccountTokenPodNodeInfo, true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceAccountTokenNodeBindingValidation, true)()
```

Comment:
This is awful and I think we should consider unconditionally plumbing through the node lister/getter from the apiserver instantiation code instead (and checking featuregates within the getter/wherever they are used).

Doing that *will* be a change to behaviour (internally) even when the featuregate is disabled however. WDYT?

Reply:
I don't think this is awful, especially when some of the gates impact the cost of the things the server runs.